### PR TITLE
User can login

### PR DESCRIPTION
**Pull request by pgaunitz**
# [PT Story](https://www.pivotaltracker.com/story/show/171561394)
## Changes proposed in this pull request:
* Adds test for User can login feature
* Added button to login form
* Added authentication model for login
## What I have learned working on this feature:
* Refresed our knowledge about user authentication using POST request  to an API

## Packages/Gems added
- Axios
-
-
## Screenshots (Only if client side story)
Screenshot from 2020-03-05 10-29-29
Screenshot from 2020-03-05 10-29-08

### CODE REVIEW

File: cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (faraznaeem) [inline cypress/fixtures/example.json]: change file name to match the content

File: cypress/integration/userCanLogin.feature.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+describe("User can log in", () => {
+    beforeEach(() => {
+      cy.server();
+      cy.visit("/");
+    });
+  
+    it("successfully", () => {
```
Comment (faraznaeem) [inline cypress/integration/userCanLogin.feature.js]: please add more info to the it blocks, eg. `user can successfully login`

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import LoginForm from "./components/LoginForm";
+
+class App extends Component {
+  state = {
+    renderLoginForm: false,
+    authenticated: false,
+    message: ""
+  };
+
+
```
Comment (faraznaeem) [inline src/App.jsx]: I would extract the login functionality to a separate component

File: src/modules/auth.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      email: email,
+      password: password
+    });
+    await storeAuthCredentials(response);
+    return { authenticated: true };
+  } 
+  catch (error) { 
+    return { authenticated: false, message: error.response.data.errors[0] };
+  }
+};
```
Comment (faraznaeem) [inline src/modules/auth.js:13]: line break

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: cypress/integration/userCanLogin.feature.js (hunk 2 above)
Comment (pgaunitz) [reply cypress/integration/userCanLogin.feature.js]: fixed

File: cypress/fixtures/example.json (hunk 1 above)
Comment (pgaunitz) [reply cypress/fixtures/example.json]: file removed

File: src/App.jsx (hunk 3 above)
Comment (pgaunitz) [reply src/App.jsx]: noted

File: src/modules/auth.js (hunk 4 above)
Comment (pgaunitz) [reply src/modules/auth.js:13]: fixed

**Comment by tochman** [comment]
Please revisit the title and PR description. Link to PT feature.

**Pull request merged**